config/keyvault/config.go

```diff
@@ -0,0 +1,205 @@
+		// https://example-keyvault.vault.azure.net/keys/example/fdf067c93bbb4b22bff4d8b7a9a56217
+		r.ExternalName.GetIDFn = getURLIDFn("keys")
+		r.References = config.References{
+			"key_vault_id": config.Reference{
+				Type:      "Vault",
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_key_vault_certificate", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetResourceNameFromIDURLFn(2)
+		// https://example-keyvault.vault.azure.net/certificates/example/fdf067c93bbb4b22bff4d8b7a9a56217
+		r.ExternalName.GetIDFn = getURLIDFn("certificates")
+		r.References = config.References{
+			"key_vault_id": config.Reference{
+				Type:      "Vault",
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_key_vault_certificate_issuer", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetResourceNameFromIDURLFn(1)
+		// https://key-vault-name.vault.azure.net/certificates/issuers/example
+		r.ExternalName.GetIDFn = getIssuerURLIDFn("certificates/issuers")
+		r.References = config.References{
+			"key_vault_id": config.Reference{
+				Type:      "Vault",
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_key_vault_managed_storage_account", func(r *config.Resource) {
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetResourceNameFromIDURLFn(1)
+		// https://example-keyvault.vault.azure.net/storage/exampleStorageAcc01
+		r.ExternalName.GetIDFn = getIssuerURLIDFn("storage")
+		r.References = config.References{
+			"key_vault_id": config.Reference{
+				Type:      rconfig.APISPackagePath + "/keyvault/v1alpha2.Vault",
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+			"storage_account_id": config.Reference{
+				Type:      rconfig.StorageAccountReferencePath,
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_key_vault_managed_hardware_security_module", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.KeyVault/managedHSMs/hsm1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.KeyVault", "managedHSMs", "name")
+	})
+
+	p.AddResourceConfigurator("azurerm_key_vault_access_policy", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.UseAsync = true
+		/* r.ExternalName = config.NameAsIdentifier
```

Comment:
Commented out code, should this be removed?

Reply:
I've left those commented out because I would like to configure `azurerm_key_vault_access_policy ` resource using `config.NameAsIdentifier`, which did not immediately work. Added a TODO for that section.